py/surveysim/nextobservation.py

```diff
@@ -61,9 +63,10 @@ def nextFieldSelector(obsplan, mjd, conditions, tilesObserved, slew, previous_ra
             if (avoidObject(dt.datetime, ra[i], dec[i]) and airMassCalculator(ra[i], dec[i], lst) < MAX_AIRMASS):
                 moondist, moonalt, moonaz = moonLoc(dt.datetime, ra[i], dec[i])
                 if ( (len(tilesObserved) > 0 and tileID[i] not in tilesObserved['TILEID']) or len(tilesObserved) == 0 ):
-                    if ( (moonalt < 0.0 and program[i] == 'DARK') or
-                         ((moonfrac < 0.2 or (moonalt*moonfrac < 12.0)) and moondist > MIN_MOON_SEP and program[i] == 'GRAY') or
-                         (program[i] == 'BRIGHT') ):
+                    if ( (moonalt < 0.0 and obsconds[i] == 1) or
+                         (moonalt >=0.0 and 
+                         (((moonfrac < 0.2 or (moonalt*moonfrac < 12.0)) and moondist > MIN_MOON_SEP and obsconds[i] == 2) or
```

Comment:
Another case where obsconds should test against a mask bit instead of a hardcoded number